bots/pr/src/main/java/org/openjdk/skara/bots/pr/BranchCommand.java

```diff
@@ -0,0 +1,133 @@
+
+import org.openjdk.skara.forge.HostedCommit;
+import org.openjdk.skara.issuetracker.Comment;
+import org.openjdk.skara.vcs.*;
+import org.openjdk.skara.jcheck.JCheckConfiguration;
+
+import java.io.PrintWriter;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.util.List;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+
+import static org.openjdk.skara.bots.common.CommandNameEnum.branch;
+
+public class BranchCommand implements CommandHandler {
+    private static final Logger log = Logger.getLogger("org.openjdk.skara.bots.pr");
+
+    private void showHelp(PrintWriter reply) {
+        reply.println("Usage: `/branch <name>`");
+    }
+
+    @Override
+    public String description() {
+        return "create a branch";
+    }
+
+    @Override
+    public String name() {
+        return branch.name();
+    }
+
+    @Override
+    public boolean allowedInCommit() {
+        return true;
+    }
+
+    @Override
+    public boolean allowedInPullRequest() {
+        return false;
+    }
+
+    @Override
+    public void handle(PullRequestBot bot, HostedCommit commit, LimitedCensusInstance censusInstance,
+            ScratchArea scratchArea, CommandInvocation command, List<Comment> allComments, PrintWriter reply) {
+        try {
+            if (!bot.integrators().contains(command.user().username())) {
+                reply.println("Only integrators for this repository are allowed to use the `/branch` command.");
+                return;
+            }
+            if (censusInstance.contributor(command.user()).isEmpty()) {
+                reply.println("To use the `/branch` command, you need to be in the OpenJDK [census](https://openjdk.org/census)"
+                        + " and your GitHub account needs to be linked with your OpenJDK username"
+                        + " ([how to associate your GitHub account with your OpenJDK username]"
+                        + "(https://wiki.openjdk.org/display/skara#Skara-AssociatingyourGitHubaccountandyourOpenJDKusername)).");
+                return;
+            }
+
+            var args = command.args();
+            if (args.isBlank()) {
+                showHelp(reply);
+                return;
+            }
+
+            var parts = args.split(" ");
+            if (parts.length > 1) {
+                showHelp(reply);
+                return;
+            }
+            var branchName = parts[0];
+
+            var localRepoDir = scratchArea.get(this)
+                    .resolve(bot.repo().name());
+            var localRepo = bot.hostedRepositoryPool()
+                               .orElseThrow(() -> new IllegalStateException("Missing repository pool for PR bot"))
+                               .materialize(bot.repo(), localRepoDir);
+            localRepo.fetch(bot.repo().authenticatedUrl(), commit.hash().toString(), true);
+
+            var remoteBranches = localRepo.remoteBranches(bot.repo().authenticatedUrl().toString());
```

Comment:
I'm not sure this will contain all branches. We have had problems with repos materialized from HostedRepositoryPool, probably a bug in there (specifically in refreshSeed). Most uses fetch explicitly after to work around it, which you do here, but this usecase is different because we need every branch to be present, not just the main commit.

Instead of working around it here, it would be better to fix HostedRepositoryPool before integrating this.

If we could trust materialize, then the fetch after wouldn't be needed.

Reply:
We actually don't need the `fetch` (we are about to push a new ref), I just left it there to "follow the pattern". I agree that we should fix the problem `HostedRepositoryPool`, but I would prefer to integrate this first, just to get if off my plate (there is no reason to hold this PR hostage over that bug 😄). 

I opted to use `HostedRepository::branches` in the latest version which hopefully will make the code a bit clearer.